Finish step2

graytraces opened the pull request:
step2 버전입니다.

graytraces committed:
Finish step2

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -7,10 +7,11 @@
     private List<PlayerCar> playerCars;
 
-    PlayRounds(int carCount){
+    PlayRounds(){}
```
굳이 이 생성자가 필요한가?

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -7,10 +7,11 @@
         playerCars = new ArrayList<>();
-        for(int i=0; i < carCount; i++){
-            playerCars.add(new PlayerCar());
+        for(int i=0; i < players.length; i++){
```
for(String player : players) {
}

위와 같이 foreach 구문으로 구현 가능함.

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -24,4 +25,31 @@ public void playSingleRounds() {
+            return winners;
+        }
+
+        return winners;
```
메소드가 너무 많은 일을 하고 있다. 다음 원칙을 지켜 메소드를 구현한다.

작게 만들어라. - 함수를 만드는 첫 번재 규칙은 '작게'다. 함수를 만드는 두 번째 규칙은 '더 작게'다.
한 가지만 해라. - 함수는 한 가지를 해야 한다. 그 한 가지를 잘 해야 한다. 그 한 가지만 해야 한다.

javajigi commented on src/main/java/racingcar/PlayRounds.java:
```diff
@@ -24,4 +25,31 @@ public void playSingleRounds() {
+            return winners;
+        }
+
+        if(maxLength == playerCar.getCarPosition().length()){
```
상태를 가지는 객체의 데이터를 꺼내려(get)하지 말고 객체에 메시지를 보내라는 원칙에 따라 구현한다.
중복 코드가 발생함.

playerCar.matchPosition(maxLength);

위와 같이 구현하고 지금 구현과 비교해 본다.

javajigi commented on src/main/java/racingcar/PlayerCar.java:
```diff
@@ -1,9 +1,16 @@
 public class PlayerCar {
+    private String carName = "";
     private String position = "-";
     static final int MOVE_MIN_VALUE = 4;
```
클래스의 구현 순서는 다음과 같다. 다음 원칙에 따라 구현한다.

```
class A {
    상수(static final) 또는 클래스 변수
    인스턴스 변수
    생성자
    메소드
}
```

javajigi commented on src/main/java/racingcar/PlayerCar.java:
```diff
@@ -1,9 +1,16 @@
     private String position = "-";
     static final int MOVE_MIN_VALUE = 4;
 
+    public PlayerCar(){};
```
이 생성자가 필요한가?

javajigi commented on src/main/java/racingcar/PlayerCar.java:
```diff
@@ -12,5 +19,9 @@ public void moveCar(int condition){
+    public String getCarName() { return carName;}
+
+    public void setCarPosition(String carPosition){ this.position = carPosition;}
+    public void setCarName(String carName){this.carName = carName;}
```
이 두개의 setter 메소드가 필요한가?

javajigi commented on src/test/java/racingcar/PlayRoundsTest.java:
```diff
@@ -0,0 +1,93 @@
+        testCar.setCarPosition("----");
+        winners = rounds.checkWinnerOrNot(maxLength, testCar, winners);
+
+        assertThat(winners.size()).isEqualTo(1);
```
테스트 메소드 하나가 너무 많은 일을 하네요.
테스트 코드 또한 한 가지에 대한 테스트 경우만 검증한다.

javajigi reviewed (changes requested):
불필요한 생성자, setter 메소드를 생성하는 경우가 없는지 의식하면 프로그래밍하면 좋을 것 같아요.

graytraces committed:
리뷰 내역 수정하였습니다.

1. 불필요한 생성자를 제거하였습니다.
2. foreach 사용하였습니다.
3. 나름 분리를 하긴 했는데 이게 맞을지는 모르겠습니다.
4. 비슷하게 구현을 하였습니다. 비교함수를 두가지 추가하였습니다.
5. static 변수를 위쪽으로 옮겼습니다.
6. 다른 불필요한 생성자를 제거하였습니다.
7. setCarName는 생성자를 통해처리했습니다. setCarPosition은 테스트에 필요하여
   public 대신에 default로 처리하였습니다.
8. 테스트에서는 리스트 길이가 2->1로 변하는 테스트를 하는 것이라서
   초기에 세팅하는 부분이 좀 길어보였던것 같습니다. 불필요한 함수호출 삭제하였습니다.
